src/autopas/LogicHandler.h

```diff
@@ -310,15 +346,15 @@ class LogicHandler {
                               static_cast<bool>(behavior & IteratorBehavior::halo) * _haloParticleBuffer.size());
     if (behavior & IteratorBehavior::owned) {
       for (auto &buffer : _particleBuffer) {
-        if (not buffer.empty()) {
-          additionalVectors.push_back(&buffer);
+        if (not buffer.isEmpty()) {
+          additionalVectors.push_back(&(buffer._particles));
         }
       }
     }
     if (behavior & IteratorBehavior::halo) {
       for (auto &buffer : _haloParticleBuffer) {
-        if (not buffer.empty()) {
-          additionalVectors.push_back(&buffer);
+        if (not buffer.isEmpty()) {
+          additionalVectors.push_back(&(buffer._particles));
```

Comment:
Maybe use 
```cpp
for (auto &cell : _particleBuffer) {
auto &buffer = cell._particles;
```
 to clarify that the buffer is a cell as you did in `deleteParticlesFromBuffer()`

Reply:
Those lines were only because before I changed the buffers to cells the code was
```cpp
for (auto &buffer : _particleBuffer ) {
```
Then, after the change, I needed to access the `_particles` member but since this (Cell) has the same interface as a `std::vector` adding this alias variable was a quick way to adapt the code with minimal changes ;)

Tbh I'm not sure I would change this here, because it would muddy the meaning of "buffer" even further (we'd call a member of an object in the buffer collection a buffer...)